spring-cloud-netflix-core/src/main/java/org/springframework/cloud/netflix/zuul/ZuulConfiguration.java

```diff
@@ -179,8 +188,40 @@ public ZuulRouteApplicationContextInitializer zuulRoutesApplicationContextInitia
 		private Map<String, ZuulFilter> filters;
 
 		@Bean
-		public ZuulFilterInitializer zuulFilterInitializer() {
-			return new ZuulFilterInitializer(this.filters);
+		public ZuulFilterInitializer zuulFilterInitializer(
+				CounterFactory counterFactory, TracerFactory tracerFactory) {
+			FilterLoader filterLoader = FilterLoader.getInstance();
+			FilterRegistry filterRegistry = FilterRegistry.instance();
+			return new ZuulFilterInitializer(this.filters, counterFactory, tracerFactory, filterLoader, filterRegistry);
+		}
+
+	}
+
+	@Configuration
+	protected static class ZuulEmptyMetricsConfiguration {
```

Comment:
There are a lot of tests that use `ZuulConfiguration`, it has inside: `ZuulFilterConfiguration`, which initializes: `ZuulFilterInitializer`. Now `ZuulFilterInitializer` depends on CounterFactory and TracerFactory beans. Both of them are setup either in `ZuulMetricsConfiguration` or in `ZuulEmptyMetricsConfiguration`. 
`ZuulMetricsConfiguration` gets setup only when class CounterService is on classpath (it creates DefaultCounterFactory). 
`ZuulEmptyMetricsConfiguration` gets setup now in any case (it creates EmptyCounterFactory and EmptyTracerFactory).
If I add @ConditionalOnMissingClass(CounterService) to `ZuulEmptyMetricsConfiguration` it won't setup, because CounterService is on classpath. 
So I end up with only `ZuulMetricsConfiguration`, which creates  DefaultCounterFactory and does not create TracerFactory. 

I can add @ConditionalOnMissingClass(CounterService) to `ZuulEmptyMetricsConfiguration`, but I will need to add setup of EmptyTracerFactory in it. Do you agree?